notebook/autogen_agent_MathChat.ipynb

```diff
@@ -0,0 +1,760 @@
+    "\n",
+    "## Requirements\n",
+    "\n",
+    "FLAML requires `Python>=3.7`. To run this notebook example, please install flaml with the [autogen] option. You also need to install `sympy` and `pydantic` for MathChat. If you want to use wolfram alpha, you also need to install `wolframalpha` package.\n",
+    "```bash\n",
+    "pip install flaml[autogen]\n",
+    "pip install sympy\n",
+    "pip install pydantic\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# %pip install flaml[autogen]==2.0.0rc1\n",
+    "# %pip install sympy\n",
+    "# %pip install pydantic\n",
+    "\n",
+    "# # Install wolframalpha if needed\n",
+    "# %pip install wolframalpha"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Set your API Endpoint\n",
+    "\n",
+    "The [`config_list_gpt4_gpt35`](https://microsoft.github.io/FLAML/docs/reference/autogen/oai/openai_utils#config_list_gpt4_gpt35) function tries to create a list of gpt-4 and gpt-3.5 configurations using Azure OpenAI endpoints and OpenAI endpoints. It assumes the api keys and api bases are stored in the corresponding environment variables or local txt files:\n",
+    "\n",
+    "- OpenAI API key: os.environ[\"OPENAI_API_KEY\"] or `openai_api_key_file=\"key_openai.txt\"`.\n",
+    "- Azure OpenAI API key: os.environ[\"AZURE_OPENAI_API_KEY\"] or `aoai_api_key_file=\"key_aoai.txt\"`. Multiple keys can be stored, one per line.\n",
+    "- Azure OpenAI API base: os.environ[\"AZURE_OPENAI_API_BASE\"] or `aoai_api_base_file=\"base_aoai.txt\"`. Multiple bases can be stored, one per line.\n",
+    "\n",
+    "It's OK to have only the OpenAI API key, or only the Azure OpenAI API key + base.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from flaml import oai\n",
+    "\n",
+    "config_list = config_list = [{\n",
+    "    'model' : 'gpt-4',\n",
+    "    'api_key': open(\"key.txt\").read().strip(),\n",
+    "}]\n",
+    "# if you open this notebook in colab, you can upload your key.txt file by click the file icon on the left panel and then choose \"upload file\".\n",
```

Comment:
why not use the `config_list_gpt4_gpt35` function as in other notebooks? The comments above and below are inconsistent with this piece of code.

Reply:
Got it